src/notifications/desktop.rs

```diff
@@ -0,0 +1,21 @@
+#[cfg(feature = "notifications")]
+use notify_rust::Notification;
+
+#[allow(unused_variables)]
+#[allow(unreachable_code)]
+pub fn notify(content: &str) -> anyhow::Result<()> {
```

Comment:
I think it would make more sense to have the function call be gated at the call site, rather than removing the implementation when the feature is disabled.

Reply:
its called multiple times, it feels weird if you have to remember each time to gate it on usage.
in this way i am not worrying if the function has to be gated or not i am adding it where i need it and if people needs it it works.

Reply:
How about:

```rust
// mod.rs

#[cfg(not(feature = "notify"))]
fn notify() { Ok(()) }

#[cfg(feature = "notify")]
fn notify() {
    impl
}
```
it looks cleaner, as you don't have conditionals within fn body

Reply:
Also, in current version, if you remove lines 7-10 it works the same and you can remove `dead_code` allows